lib/iris/tests/unit/lazy_data/test_multidim_lazy_stack.py

```diff
@@ -0,0 +1,57 @@
+# (C) British Crown Copyright 2017, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+"""Test function :func:`iris._lazy data.multidim_lazy_stack`."""
+
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests
+
+import numpy as np
+import dask.array as da
+
+from iris._lazy_data import as_lazy_data, multidim_lazy_stack
+
+
+class Test_multidim_lazy_stack(tests.IrisTest):
+    def _check(self, stack_shape):
+        vals = np.arange(np.prod(stack_shape)).reshape(stack_shape)
+        stack = np.empty(stack_shape, 'object')
+        stack_element_shape = (4, 5)
+        for index, val in np.ndenumerate(vals):
+            stack[index] = as_lazy_data(val*np.ones(stack_element_shape))
+        result = multidim_lazy_stack(stack)
+        self.assertEqual(result.shape, stack_shape + stack_element_shape)
+        self.assertIsInstance(result, da.core.Array)
+
+    def test_0d_lazy_stack(self):
+        shape = ()
+        result = self._check(shape)
+
+    def test_1d_lazy_stack(self):
+        shape = (2,)
+        result = self._check(shape)
+
+    def test_2d_lazy_stack(self):
```

Comment:
Indeed, however part of testing is checking edge cases, and I _don't know_ what would happen if I passed a 7D array to it. Evidently we don't need to test all possible dimensionalities (!), but it would be good to test something that's outside of the boundary of logical intent for the functionality being tested. This is just in case an unsafe assumption has been made about how the function will be used; "of course, _no-one_ would ever use this for more than a 2D input".